plugins/repository-gcs/src/main/java/org/elasticsearch/common/blobstore/gcs/GoogleCloudStorageBlobStore.java

```diff
@@ -259,51 +254,47 @@ void deleteBlobs(Collection<String> blobNames) throws IOException {
             deleteBlob(blobNames.iterator().next());
             return;
         }
+        final List<Storage.Objects.Delete> deletions = new ArrayList<>();
+        final Iterator<String> blobs = blobNames.iterator();
+
+        while (blobs.hasNext()) {
+            // Create a delete request for each blob to delete
+            deletions.add(client.objects().delete(bucket, blobs.next()));
+
+            if (blobs.hasNext() == false || deletions.size() == MAX_BATCHING_REQUESTS) {
+                try {
+                    // Deletions are executed using a batch request
+                    BatchRequest batch = client.batch();
```

Comment:
Yeah now that I take another look - I think `client.objects().delete` and `delete.query` need permissions. I can go back to wrapping the whole block.